data_steward/utils/participant_summary_requests.py

```diff
@@ -62,40 +83,178 @@ def get_access_token():
     return access_token
 
 
-def get_participant_data(url, headers):
+def get_participant_data(api_project_id: str,
+                         params: Dict,
+                         expected_fields: List[str] = None) -> List[Dict]:
     """
     Fetches participant data via ParticipantSummary API
 
-    :param url: the /ParticipantSummary endpoint to fetch information about the participant
-    :param headers: the metadata associated with the API request and response
+    :param api_project_id: RDR project id when PS API rests
+    :param params: the fields and their values
+    :param expected_fields: filter participants not containing any of the fields.
+        Use only if API cannot filter fields via params. If unspecified, fetches all participants
 
     :return: list of data fetched from the ParticipantSummary API
     """
+    # Base /ParticipantSummary endpoint to fetch information about the participant
+    url = BASE_URL.format(api_project_id=api_project_id)
 
     done = False
     participant_data = []
-    original_url = url
-    next_url = url
+
+    token = get_access_token()
+
+    headers = {
+        'content-type': 'application/json',
+        'Authorization': f'Bearer {token}'
+    }
+
+    session = Session()
+    retries = Retry(total=MAX_RETRIES,
+                    read=MAX_RETRIES,
+                    connect=MAX_RETRIES,
+                    backoff_factor=BACKOFF_FACTOR,
+                    status_forcelist=STATUS_FORCELIST)
+    session.mount('https://', HTTPAdapter(max_retries=retries))
+    session.mount('http://', HTTPAdapter(max_retries=retries))
 
     while not done:
-        resp = requests.get(next_url, headers=headers)
+        resp = session.get(url,
+                           headers=headers,
+                           params=params,
+                           timeout=MAX_TIMEOUT)
         if not resp or resp.status_code != 200:
-            print(f'Error: API request failed because {resp}')
+            LOGGER.warning(f'Error: API request failed because {resp}')
+            # Session has a backoff implemented, meaning a failure indicates an error with the API server, so quit
+            done = True
```

Comment:
why log a warning instead of error?  I'm assuming whatever will be using this utility will fail if we can't hit the API.  

with that thought, perhaps we should be raising an error here instead of simply exiting the loop.  maybe set `done = True` (just in case) and raise a RuntimeError after?